maas/plugins/process_check.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+#
+# Copyright 2015, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import os
+
+from maas_common import metric_bool
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+
+
+def get_process_name(pid):
+    process_path = os.path.join('/proc', pid, 'exe')
```

Comment:
This should probably be 'comm' and not 'exe'. I realize you're getting the path of the binary in addition to the binary name out of readlink() on the 'exe' file, but processes can change their names in the process table, and 'comm' will show their "common name", which is the same name you'll see in ps and related utilities. 

You can still readlink() 'exe' if you need the path for some reason, but 'comm' is the standard value to read if you intend to compare with names you get out of ps at some point, even if most of the time 'comm' is the name of the binary.